setup.py

```diff
@@ -53,7 +53,8 @@
     include_package_data=True,
     install_requires=[
         "eth-hash>=0.3.1",
-        "eth-typing>=3.0.0",
+        "eth-typing @ git+https://github.com/reedsa/eth-typing@abi-function-info-types",
```

Comment:
what will this be set to before merging?

Reply:
These changes rely on the latest in `eth-typing`. Version `4.2.0` contains the initial types for ABIs, but [breaking changes occurred](https://github.com/ethereum/eth-typing/compare/v4.2.3...main) to make required attributes for most ABI types, so I think it could warrant a major release there. cc: @fselmo @kclowes

Reply:
I'm not 100% sure if there is a question here or just an FYI, but if `eth-typing>=4.2.0` works here, we should use that, otherwise it'll be `5.0.0`?

Reply:
To clarify what will be set here: Should be `eth-typing>=4.3.1`, which includes the `ABIError` type.